lib/inspec/config.rb

```diff
@@ -137,6 +140,33 @@ def _utc_determine_backend(credentials)
       credentials[:backend] = transport_name.to_s # these are indeed stored in Train as Strings.
     end
 
+    def _utc_merge_credset(credentials, transport_name)
+      # Look for Config File credentials/transport_name/credset
+      credset_name = _utc_find_credset_name(credentials, transport_name)
+
+      if credset_name
+        credset = @cfg_file_contents.dig('credentials', transport_name, credset_name)
+        if credset
+          credentials.merge!(credset)
+        else
+          # OK, we had a target that looked like transport://something
+          # But we don't know what that something is - there was no
+          # matching credset with it.  Let train parse it.
+          credentials.merge!(Train.unpack_target_from_uri(final_options[:target]))
+        end
+      elsif final_options.key?(:target)
+        # Not sure what target looked like at all!
+        # Let train parse it.
+        credentials.merge!(Train.unpack_target_from_uri(final_options[:target]))
+      end
+    end
+
+    def _utc_find_credset_name(_credentials, transport_name)
+      return nil unless final_options[:target]
+      match = final_options[:target].match(%r{^#{transport_name}://(?<credset_name>[a-z_\-0-9]+)$})
```

Comment:
I don't think this RegEx will capture capital letters. Do we mandate the credset_name be lower case?

```
[1] pry(main)> "FOO".match(%r{(?<credset_name>[a-z_\-0-9]+)$})
=> nil
[2] pry(main)> "foo".match(%r{(?<credset_name>[a-z_\-0-9]+)$})
=> #<MatchData "foo" credset_name:"foo">
```

How about (which will catch any word, digit, or hypen):
```
[1] pry(main)> "fFo1_-".match(%r{(?<credset_name>[\w\d\-]+)$})
=> #<MatchData "fFo1_-" credset_name:"fFo1_-">
```
```suggestion
      match = final_options[:target].match(%r{^#{transport_name}://(?<credset_name>[\w\d\-]+)$})
```

Reply:
I feel like this could also use some unit tests with capitals, hypens, digits, etc.

Also, curious if URI can parse it to avoid regex all together.

Reply:
Okay, we'll make it take capital letters too.  I want to keep it as simple as possible, though.

We don't want URI to try to parse it, because URI accepts a much broader set of things than what we want to accept. http://somewhere.com/some/path is a quite valid URI; we could use URI to parse that, and then still be obliged to verify there is no path, the username is empty, the host contains no dots, etc.

This looks like parsing, but it isn't it's just pattern matching with '://' as the separator, and very specific things on each side.

If we fail matching it here - because it really is a URI - we fall back to asking Train to parse it, which is looking for URIs, and uses `URI` to parse under the hood.